src/cli/cli.cpp

```diff
@@ -2575,68 +2584,94 @@ otError Interpreter::ProcessPrefixRemove(uint8_t aArgsLength, char *aArgs[])
-        mServer->OutputFormat("%x:%x:%x:%x::/%d ", HostSwap16(config.mPrefix.mPrefix.mFields.m16[0]),
-                              HostSwap16(config.mPrefix.mPrefix.mFields.m16[1]),
-                              HostSwap16(config.mPrefix.mPrefix.mFields.m16[2]),
-                              HostSwap16(config.mPrefix.mPrefix.mFields.m16[3]), config.mPrefix.mLength);
+        mServer->OutputFormat("p");
+    }
 
-        if (config.mPreferred)
-        {
-            mServer->OutputFormat("p");
-        }
+    if (aConfig.mSlaac)
+    {
+        mServer->OutputFormat("a");
+    }
 
-        if (config.mSlaac)
-        {
-            mServer->OutputFormat("a");
-        }
+    if (aConfig.mDhcp)
+    {
+        mServer->OutputFormat("d");
+    }
 
-        if (config.mDhcp)
-        {
-            mServer->OutputFormat("d");
-        }
+    if (aConfig.mConfigure)
+    {
+        mServer->OutputFormat("c");
+    }
 
-        if (config.mConfigure)
-        {
-            mServer->OutputFormat("c");
-        }
+    if (aConfig.mDefaultRoute)
+    {
+        mServer->OutputFormat("r");
+    }
 
-        if (config.mDefaultRoute)
-        {
-            mServer->OutputFormat("r");
-        }
+    if (aConfig.mOnMesh)
+    {
+        mServer->OutputFormat("o");
+    }
 
-        if (config.mOnMesh)
-        {
-            mServer->OutputFormat("o");
-        }
+    if (aConfig.mStable)
+    {
+        mServer->OutputFormat("s");
+    }
 
-        if (config.mStable)
-        {
-            mServer->OutputFormat("s");
-        }
+    if (aConfig.mNdDns)
+    {
+        mServer->OutputFormat("n");
+    }
 
-        switch (config.mPreference)
-        {
-        case OT_ROUTE_PREFERENCE_LOW:
-            mServer->OutputFormat(" low\r\n");
-            break;
+    if (aConfig.mDp)
+    {
+        mServer->OutputFormat("D");
+    }
 
-        case OT_ROUTE_PREFERENCE_MED:
-            mServer->OutputFormat(" med\r\n");
-            break;
+    switch (aConfig.mPreference)
+    {
+    case OT_ROUTE_PREFERENCE_LOW:
+        mServer->OutputFormat(" low\r\n");
+        break;
 
-        case OT_ROUTE_PREFERENCE_HIGH:
-            mServer->OutputFormat(" high\r\n");
-            break;
+    case OT_ROUTE_PREFERENCE_MED:
+        mServer->OutputFormat(" med\r\n");
+        break;
+
+    case OT_ROUTE_PREFERENCE_HIGH:
+        mServer->OutputFormat(" high\r\n");
+        break;
+    }
+}
+
+otError Interpreter::ProcessPrefixList(void)
+{
+    otNetworkDataIterator iterator = OT_NETWORK_DATA_ITERATOR_INIT;
+    otBorderRouterConfig  config;
+
+    while (otBorderRouterGetNextOnMeshPrefix(mInstance, &iterator, &config) == OT_ERROR_NONE)
+    {
+        OutputPrefix(config);
+    }
+
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+    if (otBackboneRouterGetState(mInstance) == OT_BACKBONE_ROUTER_STATE_DISABLED)
+    {
+        if (otBackboneRouterGetDomainPrefix(mInstance, &config) == OT_ERROR_NONE)
+        {
+            OutputPrefix(config);
```

Comment:
Since the domain prefix here is not in network data, I would suggest we somehow mark this prefix as ineffective to make it easier for CLI users to tell the difference.